src/Dialect/ONNX/ONNXOps.cpp

```diff
@@ -1877,6 +1877,28 @@ LogicalResult ONNXCastOp::inferShapes() {
   return success();
 }
 
+//===----------------------------------------------------------------------===//
+// Scaler
+//===----------------------------------------------------------------------===//
+
+LogicalResult ONNXScalerOp::inferShapes() {
+  ShapedType inputType = X().getType().dyn_cast<ShapedType>();
+  if (!inputType) {
+    return emitError("Non-shaped input type");
+  }
+
+  auto getOutputType = [&inputType](Type elementType) -> Type {
+    if (inputType.hasRank()) {
+      return RankedTensorType::get(inputType.getShape(), elementType);
+    }
+    return UnrankedTensorType::get(elementType);
```

Comment:
In this path, the shape of output is unknown. You can simplify the code for this operation to:
check hasRank after ShapedType. 
getResult().setType(RankedTensorType::get(inputType.getShape(), FloatType::getF32(getContext())))